build/Versioning.props

```diff
@@ -0,0 +1,28 @@
+<Project>
+
+  <!-- Local Development -->
+  <PropertyGroup Condition="'$(CDP_BUILD_TYPE)'==''">
+    <MinorVersion>9999</MinorVersion>
+    <PatchVersion>9999</PatchVersion>
+  </PropertyGroup>
+
+  <!-- Buddy builds -->
+  <PropertyGroup Condition="'$(CDP_BUILD_TYPE)'!='' AND '$(CDP_BUILD_TYPE)'!='Official'">
+    <!-- Note that CDP_PACKAGE_VERSION_NUMERIC uses format Major.Minor.MMddyyrrrr, which causes compile error CS7034 because .NET Core doesn't allow version numbers higher than 65534. -->
+    <!-- CDP_PATCH_NUMBER is updated daily by the build system, the addition of a revision number makes the build number unique by the minute -->
+    <!-- The revision number is the number of minutes since midnight [0, 1440) -->
+    <!-- Example build number, 1.0.573.1227 -->
+    <NowTicks>$([System.DateTime]::Now.Ticks)</NowTicks>
+    <TodayTicks>$([System.DateTime]::Today.Ticks)</TodayTicks>
+    <TicksSinceMidnight>$([MSBuild]::Subtract($(NowTicks), $(TodayTicks)))</TicksSinceMidnight>
+    <MinutesSinceMidnight>$([MSBuild]::Divide($(TicksSinceMidnight), 600000000))</MinutesSinceMidnight>
+    <Floored>$([System.Math]::Floor($(MinutesSinceMidnight)))</Floored>
+    <Revision>$(Floored)</Revision>
+    <PreviewVersion>-preview-$(CDP_PATCH_NUMBER)-$(Revision)</PreviewVersion>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <Version>$(MajorVersion).$(MinorVersion).$(PatchVersion)$(PreviewVersion)</Version>
```

Comment:
`PreviewVersion` seems to be undefined for official builds.

Reply:
That's because the current version is not a preview. If it was then we could put it under `<PatchVersion` and the official build would have a preview suffix.

Reply:
This line runs for all builds, but doesn't seem to have any value for official builds. Have you tested it for an official build?

Reply:
Yes, when a variable is undefined it's value is empty string. So for official build since the current version doesn't have a preview we end up with 2.3.0

Reply:
Okay, if it is empty string, then it should work as expected. Thanks!